types/mongoose-aggregate-paginate-v2/index.d.ts

```diff
@@ -0,0 +1,90 @@
+// Type definitions for mongoose-aggregate-paginate-v2 1.0
+// Project: https://github.com/webgangster/mongoose-aggregate-paginate-v2
+// Definitions by: Linus Brolin <https://github.com/linusbrolin>
+//                 simonxca <https://github.com/simonxca>
+//                 woutgg <https://github.com/woutgg>
+//                 oktapodia <https://github.com/oktapodia>
+//                 Dongjun Lee <https://github.com/ChazEpps>
+//                 gamsterX <https://github.com/gamsterx>
+//                 knyuwork <https://github.com/knyuwork>
+//                 LiRen Tu <https://github.com/tuliren>
+//                 Alexandre Croteau <https://github.com/acrilex1>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.2
+//
+// Based on type declarations for mongoose-paginate-v2 1.3.
+
+declare module 'mongoose' {
```

Comment:
Why wrap these helper types in a namespace?

Reply:
The structure is the same as in @types/mongoose-paginate-v2. The only important part is that the `function model([...])` has to be on mongo for the plugin to work. The other interfaces could indeed be on top level. However, there seems to be no suggestion about this either in the TypeScript Handbook or in DefinitelyTyped's README.

Because mongoose-aggregate-paginate-v2 is really similar to mongoose-paginate-v2, I would also update the types for mongoose-paginate-v2 if a change in structure is required.

Reply:
Oops, I just missed that there was a function in here.